src/lambda/wizards/samDeployWizard.ts

```diff
@@ -340,14 +343,45 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
     }
 
     /**
-     * Retrieves an S3 Bucket to deploy to from the user.
+     * Prompts user to select a bucket from a regionalized list of S3 buckets
+     * Caches buckets for the entire Wizard run the first time it hits this step
+     * Allows the user to go back a step if the selected region doesn't have any buckets
      *
+     * @param selectedRegion Region selected in the last step. Used to focus S3 bucket list
      * @param initialValue Optional, Initial value to prompt with
+     * @param s3client Optional, S3 client (defaults to DefaultS3Client)
      *
-     * @returns S3 Bucket name. Undefined represents cancel.
+     * @returns S3 Bucket name. Undefined represents back.
      */
-    public async promptUserForS3Bucket(selectedRegion: string, initialValue?: string): Promise<string | undefined> {
-        const inputBox = input.createInputBox({
+    public async promptUserForS3Bucket(
+        selectedRegion: string,
+        initialValue?: string,
+        s3Client: S3Client = new DefaultS3Client(selectedRegion)
+    ): Promise<string | undefined> {
+
+        const loadingBuckets = localize(
+            'AWS.samcli.deploy.s3bucket.loading',
+            'Loading S3 buckets...',
+        )
+
+        const s3ErrorLabel = localize(
+            'AWS.samcli.deploy.s3bucket.error',
+            'There was a problem accessing S3 in region {0}',
+            selectedRegion
+        )
+
+        const s3NoBucketsLabel = localize(
+            'AWS.samcli.deploy.s3bucket.error.none',
+            'You do not have access to any S3 buckets in {0}',
```

Comment:
If the typical user has no buckets in a region, this message may cause confusion or concern.
Consider generalizing to `No buckets found in {0}`

Reply:
✔️ 

I think we had previously discussed this but I forgot the best wording to use.